kernel/src/syscall/select.rs

```diff
@@ -19,8 +19,10 @@ pub fn sys_select(
     let timeout = if timeval_addr == 0 {
         None
     } else {
-        let timeval = ctx.get_user_space().read_val::<timeval_t>(timeval_addr)?;
-        Some(Duration::from(timeval))
+        let mut timeval = ctx.get_user_space().read_val::<timeval_t>(timeval_addr)?;
+        timeval.sec += timeval.usec / 1_000_000;
+        timeval.usec %= 1_000_000;
```

Comment:
My suggestion is to do this:
```rust
impl timeval_t {
    /// Normalizes time by adding carries from microseconds to seconds.
    ///
    /// Some Linux system calls do this before checking the validity (e.g., the [poll]
    /// implementation).
    ///
    /// [poll]: https://elixir.bootlin.com/linux/v6.10.5/source/fs/select.c#L716
    pub fn normalize(&self) -> Self {
        Self {
            sec: self.sec.wrapping_add(self.usec / USEC_PER_SEC)
            usec: self.usec % USEC_PER_SEC,
        }
    }
}
```

So that all `timeval` related operations can be put in the `time/mod.rs` file, we can define some constants like `USEC_PER_SEC = 1_000_000`, `NSEC_PER_USEC = 1000` to avoid these magic numbers.

Reply:
Fixed